cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -975,8 +975,8 @@ private void generateReport(ReportGenerationRequest reportRequest, Location repo
      */
     private ExtendedReportStatus getReportStatus(Location reportIdDir) throws Exception {
       // if the report is completed but has expired, return EXPIRED as its status too
-      if (reportIdDir.append(LocationName.SUMMARY).exists()) {
-        ReportSummary reportSummary = getReportSummaryIfExists(reportIdDir);
+      ReportSummary reportSummary = getReportSummaryIfExists(reportIdDir);
```

Comment:
does this fix it? If the report is partially written, then getSummaryIfExists will try to parse an incomplete String as JSON, and we will still get an exception... It is not clear to me how the summary is written. If it is an atomic rename, then this is fine.

Reply:
It is not an atomic rewrite. Currently it does a write with the whole json string, hence when reading, it either an empty file (resulting in `null`) or the whole json obj. 

In fact, it might be better to do a rename so that there won't be any partial read.